java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueViewVector.java

```diff
@@ -0,0 +1,410 @@
+import org.apache.arrow.vector.BaseVariableWidthVector;
+import org.apache.arrow.vector.DensityAwareVector;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.NullVector;
+import org.apache.arrow.vector.UInt4Vector;
+import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.ZeroVector;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.util.CallBack;
+import org.apache.arrow.vector.util.OversizedAllocationException;
+import org.apache.arrow.vector.util.SchemaChangeRuntimeException;
+
+public abstract class BaseRepeatedValueViewVector extends BaseValueVector
+    implements RepeatedValueVector, BaseListVector {
+
+  public static final FieldVector DEFAULT_DATA_VECTOR = ZeroVector.INSTANCE;
+  public static final String DATA_VECTOR_NAME = "$data$";
+
+  public static final byte OFFSET_WIDTH = 4;
+  public static final byte SIZE_WIDTH = 4;
+  protected ArrowBuf offsetBuffer;
+  protected ArrowBuf sizeBuffer;
+  protected FieldVector vector;
+  protected final CallBack repeatedCallBack;
+  protected int valueCount;
+  protected long offsetAllocationSizeInBytes = INITIAL_VALUE_ALLOCATION * OFFSET_WIDTH;
+  protected long sizeAllocationSizeInBytes = INITIAL_VALUE_ALLOCATION * SIZE_WIDTH;
+  private final String name;
+
+  protected String defaultDataVectorName = DATA_VECTOR_NAME;
+
+  protected BaseRepeatedValueViewVector(String name, BufferAllocator allocator, CallBack callBack) {
+    this(name, allocator, DEFAULT_DATA_VECTOR, callBack);
+  }
+
+  protected BaseRepeatedValueViewVector(
+      String name, BufferAllocator allocator, FieldVector vector, CallBack callBack) {
+    super(allocator);
+    this.name = name;
+    this.offsetBuffer = allocator.getEmpty();
+    this.sizeBuffer = allocator.getEmpty();
+    this.vector = Preconditions.checkNotNull(vector, "data vector cannot be null");
+    this.repeatedCallBack = callBack;
+    this.valueCount = 0;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  @Override
+  public boolean allocateNewSafe() {
+    boolean dataAlloc = false;
+    try {
+      allocateBuffers();
+      dataAlloc = vector.allocateNewSafe();
+    } catch (Exception e) {
+      clear();
+      return false;
+    } finally {
+      if (!dataAlloc) {
+        clear();
+      }
+    }
+    return dataAlloc;
+  }
+
+  private void allocateBuffers() {
+    offsetBuffer = allocateBuffers(offsetAllocationSizeInBytes, true);
+    sizeBuffer = allocateBuffers(sizeAllocationSizeInBytes, false);
+  }
+
+  private ArrowBuf allocateBuffers(final long size, boolean isOffSet) {
+    final int curSize = (int) size;
+    ArrowBuf buffer = allocator.buffer(curSize);
+    buffer.readerIndex(0);
+    buffer.setZero(0, buffer.capacity());
+    if (isOffSet) {
+      offsetAllocationSizeInBytes = curSize;
+    } else {
+      sizeAllocationSizeInBytes = curSize;
+    }
```

Comment:
no flag. just set `offsetAllocationSizeInBytes` and `sizeAllocationSizeInBytes` inside `allocateBuffers`.

Reply:
And I don't think we even have to set that, because we are setting the same value to itself.